morebits.css

```diff
@@ -304,3 +349,39 @@ th.morebits-mbox-text, td.morebits-mbox-text {
     padding: 0.25em 0.9em;
     width: 100%;
 }
+
+/* Twinkle config */
+#twinkle-config {
+	border: 1px solid #666;
+	background: var(--morebits-bgcolor-dialog, #F0F8FF);
+	color: inherit;
+}
+
+#twinkle-config label {
+	color: var(--color-base, #202122);
+}
+
+#twinkle-config-titlebar {
+	background: var(--morebits-bgcolor-titlebar, #BCCADF);
+	color: inherit;
+}
+
+#twinkle-config-buttonpane {
+	background: var(--morebits-bgcolor-titlebar, #BCCADF);
+	color: inherit;
+	padding: 0.5em;
+}
+
+.twinkle-config-helptip {
+	font-size: 90%;
+	color: var(--color-subtle, #54595d);
+}
+
+/* Override select2 silliness */
```

Comment:
select2 dropdowns look broken:

<img width="700" alt="Screenshot 2024-10-20 at 1 12 42 AM" src="https://github.com/user-attachments/assets/dddfdd0d-b889-4fbc-a9e9-4ae774fbb1f1">

Reply:
I think we might need to fork select2, and make our own custom css styles for dark mode. Applying this css hotfix to morebits.css is only likely to cause problems down the road.

Reply:
My main concern with this solution is because it's not specific to Twinkle, it may affect styles for content outside of Twinkle. That's why I suggested changing the select2 library's css directly.

Reply:
All select2 CSS we have today are added via JS in the module code ([example](https://github.com/wikimedia-gadgets/twinkle/blob/34259325a9043b246b7d8cd5ae338d09ee244415/modules/twinklewarn.js#L1499)) so that they are applied only when that Twinkle dialog is opened. We could do the same here, though it will still affect non-Twinkle content when these modules (warn/xfd/block) are opened. I think it's a reasonable trade-off.

There's no way around that short of creating a [custom select2 theme](https://select2.org/appearance#themes), which involves duplicating a lot of CSS of the default theme. 

Maintaining a fork of the entire library is not something we have bandwidth for. Even if we had, select2 fits in poorly with MediaWiki's interfaces and we should ideally move away from it (in favour of integrations with native libraries like Codex or OOUI), not deepen our involvement with it.

Reply:
I found a way to avoid impacting non-Twinkle content. Turns out we can use the `theme` field to hack in an extra class on the select2-container. By setting `theme: 'default select2-morebits'`, the default theme styles get applied, and `select2-morebits` class also appears on the container using which we can target the styles: 
```css
.select2-morebits,
.select2-morebits .select2-results,
.select2-morebits .select2-results__option[aria-selected=true],
.select2-morebits .select2-selection__rendered,
.select2-morebits .select2-selection--multiple,
.select2-morebits .select2-selection__choice {
    background: var(--background-color-base, #fff) !important;
    color: inherit !important;
}
```

Reply:
I'm not that familiar with select2, where do I apply the `theme` field, so it applies to all select2 objects? I tried doing so by changing the default in lib/select2.min.js, but that didn't work.

Reply:
Actually I found a better strategy:

```css
.morebits-dialog ~ .select2-container:first-of-type > .select2-dropdown {
    background: var(--background-color-base, #fff) !important;
    color: inherit !important;
}
```

This CSS only matches the first select2-container after morebits-dialog, therefore only matching select2 twinkle content.

Reply:
> I'm not that familiar with select2, where do I apply the `theme` field

In the `.select2({...})` initialization at each place where select2 is used (in twinklewarn.js, twinkleblock.js and twinklexfd.js).

> so it applies to all select2 objects?

Even if it was possible to apply it in one place, that would mean any non-Twinkle uses also get the styles, which is what you set out to avoid in the first place?

> Actually I found a better strategy:

That's a pretty complex selector. I think using `.select2-morebits` in place of `.morebits-dialog ~ .select2-container:first-of-type` is cleaner and less error-prone.

Reply:
Okay, that should be done now.